### PR TITLE
Unified combination command - isDefault, prices and stock

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This follow the unification process of Combination command, this PR handles the prices and stock sub-domains along with the isDefault field
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| Related PRs       | ~
| How to test?      | CI tests green, only behat modifications in this PR
| Possible impacts? | ~

### CODE REVIEW

File: src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php
Comment (zuk3975) [inline src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php:98]: todo should be resolved after this pr merge https://github.com/PrestaShop/PrestaShop/pull/30320

File: src/Adapter/Product/Combination/CommandHandler/UpdateCombinationStockAvailableHandler.php
Comment (zuk3975) [inline src/Adapter/Product/Combination/CommandHandler/UpdateCombinationStockAvailableHandler.php:54]:
```suggestion
    
    /**
```

File: src/Core/Domain/Product/Combination/Command/UpdateCombinationCommand.php
Comment (zuk3975) [inline src/Core/Domain/Product/Combination/Command/UpdateCombinationCommand.php:88]: :+1:

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml]:
```suggestion
```

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml]: lint warning
`comment not indented like content`

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml]: lint warning same here

File: src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php
Comment (lartist) [inline src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php:99]: I think we should pass the context shop id to the command instead using \Context class in the Handler.

File: src/PrestaShopBundle/Resources/config/services/adapter/product.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product.yml:405]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:97]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:117]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:137]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:218]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:222]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:224]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:227]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_combination.yml:230]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_stock.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_stock.yml:8]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/adapter/product/_stock.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/adapter/product/_stock.yml:31]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:153]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml]: Need autowire true?

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml:153]: I don't think so since it has no arguments, if some are added the autowire can be added when needed then

File: src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml
Comment (jolelievre) [reply src/PrestaShopBundle/Resources/config/services/core/form/command_builder.yml]: Same as the other, no arguments so autowiring not needed for now

File: src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php
Comment (jolelievre) [reply src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php:99]: Yeah I agree, but this is part of a refacto that is done in another PR here https://github.com/PrestaShop/PrestaShop/pull/30320

So depending on which one is merged first I'll fix this in this PR or in a future one, so for now I left the same todo comment as in the initial handler

Comment (lartist) [reply src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php:99]: Ok

**Review by lartist — DISMISSED** [review]

**Comment by zuk3975** [comment]
sanity tests failing, lets reopen and see if it was random

**Review by zuk3975 — DISMISSED** [review]

**Comment by jolelievre** [comment]
DO NOT MERGE THIS PR before the [merge PR](https://github.com/PrestaShop/PrestaShop/pull/30399) is merged

**EDIT**- all good, that one is merged :heavy_check_mark:

**Review by zuk3975 — APPROVED** [review]

**Review by lartist — APPROVED** [review]

File: src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php
Comment (zuk3975) [reply src/Adapter/Product/Combination/CommandHandler/UpdateCombinationHandler.php:98]: added into todo reminder issue to make sure https://github.com/PrestaShop/PrestaShop/issues/30304#issue-1451647434

**Pull request merged**
